webpack.config.js

```diff
@@ -0,0 +1,219 @@
+                // Those less than 1024 bytes are automatically encoded in the CSS - see `_test-background-images.scss`
+                // the publicPath matches the path from the compiled css to the cssBackgrounds file
+                test: /\.(svg|jpg|png)$/,
+                include: path.resolve(
+                    `./${projectRoot}/static_src/images/cssBackgrounds/`,
+                ),
+                use: {
+                    loader: 'url-loader',
+                    options: {
+                        fallback: 'file-loader',
+                        name: '[name].[ext]',
+                        outputPath: 'images/cssBackgrounds/',
+                        publicPath: '../images/cssBackgrounds/',
+                        limit: 1024,
+                    },
+                },
+            },
+        ],
+    },
+    // externals are loaded via base.html and not included in the webpack bundle.
+    externals: {
+        // gettext: 'gettext',
+    },
+};
+
+/*
+  If a project requires internationalisation, then include `gettext` in base.html
+    via the Django JSi18n helper, and uncomment it from the 'externals' object above.
+*/
+
+const webpackConfig = (environment, argv) => {
+    const isProduction = argv.mode === 'production';
+
+    options.mode = isProduction ? 'production' : 'development';
+
+    if (!isProduction) {
+        // https://webpack.js.org/configuration/stats/
+        const stats = {
+            // Tells stats whether to add the build date and the build time information.
+            builtAt: false,
+            // Add chunk information (setting this to `false` allows for a less verbose output)
+            chunks: false,
+            // Add the hash of the compilation
+            hash: false,
+            // `webpack --colors` equivalent
+            colors: true,
+            // Add information about the reasons why modules are included
+            reasons: false,
+            // Add webpack version information
+            version: false,
+            // Add built modules information
+            modules: false,
+            // Show performance hint when file size exceeds `performance.maxAssetSize`
+            performance: false,
+            // Add children information
+            children: false,
+            // Add asset Information.
+            assets: false,
+        };
+
+        options.stats = stats;
+
+        // Create JS source maps in the dev mode
+        // See https://webpack.js.org/configuration/devtool/ for more options
+        options.devtool = 'inline-source-map';
+
+        // See https://webpack.js.org/configuration/dev-server/.
+        options.devServer = {
+            // Enable gzip compression for everything served.
+            compress: true,
+            static: false,
+            host: '0.0.0.0',
+            // When set to 'auto' this option always allows localhost, host, and client.webSocketURL.hostname
+            allowedHosts: 'auto',
+            port: 3000,
```

Comment:
Issue: This port should be exposed in the `docker-compose.yml`.

Reply:
I've just spotted https://git.torchbox.com/internal/wagtail-kit/-/blob/main/layers/base/docker/docker-compose-frontend.yml?ref_type=heads in kit - do we need that file in this repo too?

Reply:
Yes - that lets the frontend run properly in docker.

Reply:
@RealOrangeOne  I've just added the port to docker-compose.yml for now - is that sufficient for this project? I never run the fe tooling in docker myself so would be good if you could test if you have a chance.

Reply:
Hmm, http://localhost:3000/ doesn't work at all so I'm guessing this isn't correct. Any hints to get this working?

Reply:
@helenb did you get this working?